test/hotspot/jtreg/runtime/logging/AsyncDeathTestNonProduct.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8349755
+ * @summary Perform recursive logging in async UL and do NOT observe a crash (as opposed to AsyncDeathTestDebug)
+ * @requires vm.flagless
+ * @requires !vm.debug
+ * @library /test/lib
+ * @modules java.base/jdk.internal.misc
+ *          java.management
+ * @run driver AsyncDeathTestDebug
+ */
+
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.process.OutputAnalyzer;
+
+public class AsyncDeathNonProduct {
```

Comment:
Did you mean to call this `AsyncDeathNonDebug` ??

You could just have one test file and two @test sections: one for debug and one for non-debug, then adapt the logic accordingly.  But this non-debug version isn't verifying the correct execution of recursive logging in product mode, because there is no actual recursive logging being performed - this test isn't doing anything. You would need some product logging to actually verify that but then anything that did `-Xlog:all` would crash. So perhaps just delete this version of the test.

Reply:
I meant to call it `NonProduct`, and the rest of your message is confusing to me so I've probably misunderstood what my test does and what `NOT_PRODUCT` means :-).

I've got:

```c++
NOT_PRODUCT(log_debug(deathtest)("Induce a recursive log for testing");)
```

I thought `NON_PRODUCT` means that this is included regardless of  `assert`s being enabled, but will not be shipped in a product build of the JDK. That's what I want to do, as that let's me test recursive logging without any asserts being triggered.

I'll look into the test section suggestion.

Reply:
I've misunderstood your intention, but we only run tests on debug and product and your `@requires !vm.debug` means the test will run on product and it will do nothing because the log statements will not be there.

Reply:
Aha, I see. Well, I think that I can re-work the code that let's us test it to accommodate for that.